src/interpreter.js

```
@@ -0,0 +1,141 @@
+/*
+* Author: Jonas Finnemann Jensen
+* Github: https://github.com/jonasfj
+*/
+let PrattParser = require('./prattparser');
+
+let parseList = (ctx, separator, terminator) => {
+  let list = [];
+  if (!ctx.try(terminator)) {
+    do {
+      list.push(ctx.parse());
+    } while (ctx.try(separator));
+    ctx.require(terminator);
+  }
+  return list;
+};
+
+let parseObject = (ctx) => {
+  let obj = {};
+  if (!this.try('}')) {
+    do {
+      let k = this.require('id', 'string');
+      this.require(':');
+      let v = this.parse();
+      obj[k] = v;
+    } while (this.try(','));
+    this.require('}');
+  }
+  return obj;
+};
+
+let parseKey = (left, token, ctx) => {
+  if (left instanceof Array) {
+    let begin, end;
+    if (ctx.try(':')) {
+      begin = 0;
+      if (ctx.try(']')) {
+        end = left.length;
+      } else {
+        end = ctx.parse();
+        ctx.require(']');
+      }
+    } else {
+      begin = ctx.parse();
+      if (ctx.try(':')) {
+        if (ctx.try(']')) {
+          end = left.length;
+        } else {
+          end = ctx.parse();
+          ctx.require(']');
+        }
+      } else {
+        ctx.require(']');
+      }
+    }
+    if (begin !== undefined && end !== undefined) {
+      return left.slice(begin, end);
+    }
+    return left[begin];
+  } else {
+    let result = left[ctx.parse()];
+    ctx.require(']');
+    return result;
+  }
+};
+
+let compareNumbers = (left, operator, right) => {
+  switch (operator) {
+    case '>=': return left >= right;
+    case '<=': return left <= right;
+    case '>': return left > right;
+    case '<': return left < right;
+    default: return left < right;
```

Let's use default to throw a syntax error for unknown operator...